wfcGen.py

```diff
@@ -0,0 +1,264 @@
+import random
+
+from PIL import Image
+
+
+res = 3
+cropShift = False
+cells = {}
+cropped = []
+for ins in range(4):
+    field = Image.open("image.png")
+    if cropShift:
+
+        field = field.rotate(90*ins)
+
+        SX = field.size[0]
+        SY = field.size[1]
+
+        copy = Image.new("RGB", (SX*2, SY*2))
+        copy.paste(field, (0, 0))
+        copy.paste(field, (SX, 0))
+        copy.paste(field, (0, SY))
+        copy.paste(field, (SX, SY))
+        field = copy
+
+        for y in range(SY):
+            cropped[ins].append([])
+            for x in range(SX):
+                cropped[ins][y].append(field.crop((x, y, x+res, y+res)))
+
+        cropped[ins] = [[field.crop((x, y, x+res, y+res)) for x in range(SX)] for y in range(SY)]
+
+    else:
+
+        field = field.rotate(90*ins)
+        SX = field.size[0]//res*res
+        SY = field.size[1]//res*res
+        field = field.crop((0, 0, SX, SY))
+
+        cropped[ins] = [[field.crop((x*res, y*res, x*res+res, y*res+res)) for x in range(SX//res)] for y in range(SY//res)]
+
+    lcy = len(cropped[ins])
+    lcx = len(cropped[ins][0])
+
+    print(lcy, lcx)
+
+
+
+    # [left, top, right, bottom]uwu
+    # u will newer need that
+
+    for y in range(lcy):
+        for x in range(lcx):
+            flag = -1
+            for key in cells.keys():
+                if cropped[ins][y][x] == cells[key]["img"]:
+                    flag = key
+
+            if flag != -1:
+                cells[flag]["nbg"]["0"].append(cropped[ins][y][x-1-(res-1)*int(cropShift)])
+                cells[flag]["nbg"]["1"].append(cropped[ins][y-1-(res-1)*int(cropShift)][x])
+                cells[flag]["nbg"]["2"].append(cropped[ins][y][(x+1+(res-1)*int(cropShift))%lcx])
+                cells[flag]["nbg"]["3"].append(cropped[ins][(y+1+(res-1)*int(cropShift))%lcy][x])
+            else:
+                cells[str(len(cells.keys()))] = {}
+                cells[str(len(cells.keys())-1)]["nbg"] = {}
+                cells[str(len(cells.keys())-1)]["nbg"]["0"] = [cropped[ins][y][x-1-(res-1)*int(cropShift)]]
+                cells[str(len(cells.keys())-1)]["nbg"]["1"] = [cropped[ins][y-1-(res-1)*int(cropShift)][x]]
+                cells[str(len(cells.keys())-1)]["nbg"]["2"] = [cropped[ins][y][(x+1+(res-1)*int(cropShift))%lcx]]
+                cells[str(len(cells.keys())-1)]["nbg"]["3"] = [cropped[ins][(y+1+(res-1)*int(cropShift))%lcy][x]]
+                cells[str(len(cells.keys())-1)]["img"] = cropped[ins][y][x]

```

Comment:
1. Why `str(len(cells.keys()))` rather than just `len(cells.keys())`?
2. Why do you create an empty dictionary and then make four additions to it? You can create the dictionary with these values already there. If you care about the nice multiline formatting (which I _would_ care about, if I were you), there is no issue with that as well: you can write a multi-line declaration of a dictionary, such as
```python
d = {
    1: "aba",
    2: "caba",
    3: "abacaba",
}
```